src/app/components/post-detail/post-detail.component.ts

```diff
@@ -17,30 +18,51 @@ export class PostDetailComponent implements OnInit {
 
   private isMyPost: boolean = false;
 
-  constructor(private postService: PostService, private route: ActivatedRoute, private authService: AuthService) { }
+  constructor(private postService: PostService,
+              private route: ActivatedRoute, private authService: AuthService,
+              private loc: Location) { }
 
   ngOnInit() {
       this.route.params.subscribe(params => {
       this.postService.getPost(params['_id'])
-        .then(post => this.post = post);
+        .then(post => this.initPostStatus(post));
+
     });
+
+
+
+  }
+  initPostStatus(post: Post){
+    this.post = post;
     if (this.authService.userLoggedIn) {
-      if (this.authService.userLoggedIn.user.userID = this.post.author._id) {
-        this.isMyPost = true;
-      } else {
+
+        if (this.authService.userLoggedIn.user.userID === this.post.author._id) {
+            this.isMyPost = true;
+        } else {
+            this.isMyPost = false;
+        }
+    }
+    else {
         this.isMyPost = false;
-      }
-    } else {
-      this.isMyPost = false;
     }
   }
 
+  Delete(){
+    this.postService.delete(this.post._id)
+    alert('Post Deleted!')
+    this.loc.back()
```

Comment:
Careful with using `Location`. We should instead navigate back to the posts view by injecting `Router` and calling `router.navigate['/posts']`. If a user navigates to a specific post from an external link, this will error. Also let's not use browser alerts - they generally come off as intrusive (from a user perspective). They're also unstylable!

Reply:
roger roger